queries/fortran/highlights.scm

```diff
@@ -1,114 +1,272 @@
-(identifier) @variable
-(string_literal) @string
-(number_literal) @number
-(boolean_literal) @boolean
-(comment) @comment
+; Preprocs
+
+(preproc_file_line) @preproc
+
+; Namespaces
+
+(program_statement
+  (name) @namespace)
+
+(end_program_statement
+  (name) @namespace)
+
+(module_statement
+  (name) @namespace)
+
+(end_module_statement
+  (name) @namespace)
+
+(submodule_statement
+  (name) @namespace)
+
+(end_submodule_statement
+  (name) @namespace)
+
+; Includes
 
 [
-  (intrinsic_type)
-  "pointer"
+  "import"
+  "include"
+  "use"
+] @include
+
+(import_statement
+  ","
+  ["all" "none"] @keyword)
+
+; Attributes
+
+[
+  (none)
+  "implicit"
+  "intent"
+] @attribute
+
+(implicit_statement
+  "type" @attribute)
+
+; Keywords
+
+[
+  "attributes"
+  "associate"
+  "block"
+  "class"
+  "contains"
+  "default"
+  "dimension"
+  "endassociate"
+  "endenum"
+  "endinterface"
+  "endmodule"
+  "endselect"
+  "endsubmodule"
+  "enum"
+  "enumerator"
+  "equivalence"
+  "extends"
+  "interface"
+  "intrinsic"
+  "non_intrinsic"
+  "module"
+  "submodule"
+  "namelist"
+  "parameter"
+  "quiet"
+  "rank"
+  "save"
+  "selectcase"
+  "selectrank"
+  "selecttype"
+  "sequence"
+  "stop"
+  "target"
+] @keyword
+
+[
+  (default)
+] @keyword
+
+; Types
+
+[
+  (type_name)
 ] @type
 
 [
+  (intrinsic_type)
+] @type.builtin
+
+; Typedefs
+
+[
+  "classis"
+  "endtype"
+  "type"
+  "typeis"
+] @type.definition
+
+; Qualifiers
+
+[
+  "abstract"
   "allocatable"
-  "attributes"
+  "automatic"
+  "constant"
+  "contiguous"
+  "data"
+  "deferred"
   "device"
-  "dimension"
+  "external"
+  "family"
+  "final"
+  "generic"
   "global"
   "grid_global"
   "host"
+  "initial"
+  "local"
+  "local_init"
+  "managed"
+  "nopass"
+  "non_overridable"
   "optional"
+  "pass"
+  "pinned"
+  "pointer"
   "private"
+  "property"
+  "protected"
   "public"
+  "shared"
+  "static"
+  "texture"
   "value"
+  "volatile"
+  (procedure_qualifier)
 ] @type.qualifier
 
 [
-  "endtype"
-  "type"
-] @type.definition
-
-[
+  "common"
   "in"
   "inout"
   "out"
 ] @storageclass
 
-[
-  "contains"
-  "import"
-] @include
+; Labels
 
 [
-  (none)
-  "implicit"
-  "intent"
-] @attribute
+  "goto"
```

Comment:
I'm not sure if that is what `@label` is supposed to be. Does C this? Actually I would expect  `@label` for what comes after `@goto`

Reply:
Oops, I mistook what `@label` was for - fixed